data/data-gen/generateDescriptions.js

```diff
@@ -0,0 +1,59 @@
+const fs = require('fs');
+const { convertObjToCSV, generatePhrase, generateSentence } = require('./utils');
+
+/**
+ * Write numRecords number of descriptions, each with title & description fields, to CSV file at outDir
+ * @param {Number} numRecords: number between 1 and 10 million, inclusive
+ * @param {String} outDir: directory to write resulting CSV file to
+ * @returns {Promise->Boolean}
+ */
+exports.generateDescriptions = async (numRecords, outDir) => {
+  const stream = fs.createWriteStream(`${outDir}/descriptions.csv`);
+
+  // Declare listeners
+  stream.on('error', err => {
+    throw err;
+  });
+
+  stream.on('close', () => {
+    console.log(`\nDescriptions file generated. Check ${outDir}\\descriptions.csv.`);
+  });
+
+  stream.on('open', () => {
+    console.log(`\nWriting ${numRecords} descriptions to file. For 10M records, this will take around 4 minutes. Please wait...`);
```

Comment:
Love the comments/listeners! Very helpful to let the user know the expected time.